diff --git a/client/src/components/forms/Sphere.vue b/client/src/components/forms/Sphere.vue
index 0c8b6d32..4e480875 100644
--- a/client/src/components/forms/Sphere.vue
+++ b/client/src/components/forms/Sphere.vue
@@ -39,7 +39,6 @@
         name="type"
         placeholder="Select the type"
         :options="[
-          { label: 'None', value: '' },
           { label: 'Topic', value: 'topic' },
           { label: 'Location', value: 'location' }
         ]"
@@ -48,7 +47,7 @@
     </div>
 
     <SubmitButton
-      :resetTrigger="errorMessage && 0"
+      :resetTrigger="errorMessage"
       className="w-full mt-4"
     >
       Create Sphere
@@ -81,8 +80,11 @@ const onSubmit = async () => {
   errorMessage.value = '';
   try {
     const formData = new FormData(form.value);
+    const type = formData.get('type');
+    if (type != 'topic' && type != 'location') {
+      throw new Error('Please, select a type');
+    }
     const result = await createSphere(formData);
-    //create relation
 
     return router.push(`/sphere/${result.id}`);
   } catch (error) {
diff --git a/client/src/composables/sphereProvider.ts b/client/src/composables/sphereProvider.ts
index 5198d044..1aa8c927 100644
--- a/client/src/composables/sphereProvider.ts
+++ b/client/src/composables/sphereProvider.ts
@@ -56,7 +56,7 @@ export function sphereProvider() {
 
       setTimeout(() => resolve(results), 300);
     });
-  };
+  }
 
   function incrementLastChar(str: string) { //utils
     const last = str.slice(-1);
diff --git a/client/src/views/sphere/DetailView.vue b/client/src/views/sphere/DetailView.vue
index 753b3deb..6a4f25db 100644
--- a/client/src/views/sphere/DetailView.vue
+++ b/client/src/views/sphere/DetailView.vue
@@ -1,5 +1,6 @@
 <template>
   <RelationProvider>
+    <!--
     <Banner v-if="space == 'local'">
       You are viewing the local version of {{ sphere?.title }}.
       <b
@@ -8,6 +9,7 @@
       > Switch to global
       </b>
     </Banner>
+    -->
 
     <SubNav
       :initialTab="tab"
diff --git a/client/src/views/sphere/InfoView.vue b/client/src/views/sphere/InfoView.vue
index 7fc0c900..1377565a 100644
--- a/client/src/views/sphere/InfoView.vue
+++ b/client/src/views/sphere/InfoView.vue
@@ -97,4 +97,4 @@ const pluginSlots = computed(() => (
 onMounted(async () => {
   tab.value = 'Info';
 });
-</script>
\ No newline at end of file
+</script>
diff --git a/client/src/views/sphere/SettingsView.vue b/client/src/views/sphere/SettingsView.vue
index 0925a0fc..e501f24e 100644
--- a/client/src/views/sphere/SettingsView.vue
+++ b/client/src/views/sphere/SettingsView.vue
@@ -77,4 +77,4 @@ import { topicRelations, locationRelations } from '@/assets/relationKeys';
 
 const { sphere } = useSphere();
 const { slots } = usePlugins();
-</script>
\ No newline at end of file
+</script>
